client/src/pages/CheckoutPage.js

```diff
@@ -0,0 +1,288 @@
+import React, { useContext, useEffect } from "react";
+import { Paper, Typography, Grid } from "@material-ui/core";
+import { useTheme, makeStyles } from "@material-ui/core/styles";
+import { Elements } from "@stripe/react-stripe-js";
+import FlightTakeoffIcon from "@material-ui/icons/FlightTakeoff";
+import { loadStripe } from "@stripe/stripe-js";
+import FlightLandIcon from "@material-ui/icons/FlightLand";
+import ApartmentIcon from "@material-ui/icons/Apartment";
+import { ShoppingCartContext } from "../components/ShoppingCartContext";
+import axios from "axios";
+import DirectionsCarIcon from "@material-ui/icons/DirectionsCar";
+import CheckoutForm from "../components/CheckoutForm";
+
+const checkoutStyles = makeStyles((theme) => ({
+  cartSummaryContainer: {
+    width: "40%",
+    borderLeft: `1px solid ${theme.palette.primary.light}`,
+    padding: "2rem",
+    textAlign: "center",
+  },
+  flightContainer: {
+    margin: "1rem 0.5rem 1rem 0.5rem",
+  },
+  destinationText: {
+    marginTop: "1rem",
+    textAlign: "center",
+  },
+  checkoutContainer: {
+    width: "60%",
+    backgroundColor: "#ebf0f7",
+  },
+  checkoutPaper: {
+    marginTop: "2rem",
+    padding: "1.5rem 1rem 1.5rem 1rem",
+    width: "80%",
+  },
+  pageContainer: {
+    height: "calc(100vh - 64px)",
+  },
+  carText: {
+    textAlign: "center",
+  },
+  subTotalContainer: {
+    borderTop: `1px solid ${theme.palette.primary.light}`,
+    marginTop: `1rem`,
+    paddingTop: "1rem",
+  },
+  priceText: {
+    marginLeft: "1rem",
+  },
+  icons: {
+    paddingRight: "8px",
+    marginBottom: "-7px",
+  },
+}));
+
+function CheckoutPage() {
+  const theme = useTheme();
+  const classes = checkoutStyles(theme);
+
+  const [shoppingCart, setShoppingCart] = useContext(ShoppingCartContext);
+
+  let car,
+    hotel,
+    flight = false;
+
+  useEffect(() => {
+    const fetchCart = async () => {
+      try {
+        await axios.get("/shopping-cart/get-cart").then((data) => {
+          setShoppingCart(data.data.cart);
+        });
+      } catch (e) {
+        console.log(e);
+      }
+    };
+
+    fetchCart();
+  }, []);
+
+  let prices = [];
+
+  for (let i = 0; i < shoppingCart.length; i++) {
+    if (shoppingCart[i].type === "car") {
+      car = shoppingCart[i];
+    }
+    if (shoppingCart[i].type === "hotel") {
+      hotel = shoppingCart[i];
+    }
+    if (shoppingCart[i].type === "flight") {
+      flight = shoppingCart[i];
+    }
+    prices.push(parseInt(shoppingCart[i].price));
+  }
```

Comment:
Can't we move this inside the useEffect? and just use `data.data.cart`

Reply:
Yes that actually makes a lot more sense! Will fix that.

Reply:
So i tried it and got this warning message. Should i just go ahead and keep it out of use effect? 

`Assignments to the 'car' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect.` 

I tried it out and I do lose the variable so nothing shows up in the cart.

Reply:
Oh yeah that's a good point. Let's leave it for now. I think the actual solution would be something like change the cart so that instead of an array, it's an object containing car/flight/hotel, then you can just directly reference it instead of needing more variables